Source/OIDAuthState.m

```diff
@@ -159,8 +159,13 @@ @implementation OIDAuthState {
                                                 callback(authState, tokenError);
                                }];
                              } else {
-                               // implicit or hybrid flow (hybrid flow assumes code is not for this
-                               // client)
+                               // hybrid flow (code id_token). Two possible cases:
+                               // 1. The code is not for this client, ie. will be sent to a
+                               //    webservice that performs the id token verification and token
+                               //    exchange
+                               // 2. The code is for this client and, for security reasons, the
+                               //    application developer must verify the id_token signature and
+                               //    c_hash before calling the token endpoint
```

Comment:
add a note perhaps… "for that reason, the code exchange is not performed automatically" ?

Reply:
I assume that for hybrid you won't use this convenience initializer, is that how you've implemented it?

Reply:
We actually do use this convenience initialiser, and pick up the id_token up from authState.lastAuthorizationResponse.idToken - I can't say that was a deliberate decision though. Does using it for the hybrid flow worry you in any way?

Reply:
Not it doesn't worry me, but I think we should be more explicit that hybrid is officially "supported" by this method.

Can you enhance the method doc? Currently it reads:

> Convenience method to create a @c OIDAuthState by presenting an authorization request
        and performing the authorization code exchange in the case of code flow requests.

It doesn't exclude hybrid but doesn't exactly tell the user much either. Documenting the behavior for hybrid would be good (i.e. it doesn't do the code exchange, you're expected to validate the token then do the exchange yourself, and call the updateWithTokenResponse method).

Reply:
Yes. That makes sense. I'll push this change shortly, thanks.